solidity/random-beacon/contracts/ReimbursmentPool.sol

```diff
@@ -0,0 +1,132 @@
+    ///      contract resposibility to make sure it can receive ETH.
+    /// @dev Only authorized contracts are allowed calling this function.
+    /// @param gasSpent Gas spent on a transaction that needs to be reimbursed.
+    /// @param receiver Address where the reimbursment is sent.
+    function refund(uint256 gasSpent, address receiver) external nonReentrant {
+        require(
+            isAuthorized[msg.sender],
+            "Contract is not authorized for a refund"
+        );
+        require(receiver != address(0), "Receiver's address cannot be zero");
+
+        uint256 gasPrice = tx.gasprice < maxGasPrice
+            ? tx.gasprice
+            : maxGasPrice;
+
+        uint256 refundAmount = (gasSpent + staticGas) * gasPrice;
+
+        /* solhint-disable avoid-low-level-calls */
+        // slither-disable-next-line low-level-calls,unchecked-lowlevel
+        receiver.call{value: refundAmount}("");
+        /* solhint-enable avoid-low-level-calls */
+    }
+
+    /// @notice Authorize a contract that can interact with this reimbursment pool.
+    ///         Can be authorized by the owner only.
+    /// @param _contract Authorized contract.
+    function authorize(address _contract) external onlyOwner {
+        isAuthorized[_contract] = true;
+    }
+
+    /// @notice Unauthorize a contract that was previously authorized to interact
+    ///         with this reimbursment pool. Can be unauthorized by the
+    ///         owner only.
+    /// @param _contract Authorized contract.
+    function unauthorize(address _contract) external onlyOwner {
+        delete isAuthorized[_contract];
+    }
+
+    /// @notice Setting a static gas cost for executing a transaction. Can be set
+    ///         by the owner only.
+    /// @param _staticGas Static gas cost.
+    function setStaticGas(uint256 _staticGas) external onlyOwner {
+        staticGas = _staticGas;
+
+        emit StaticGasUpdated(_staticGas);
+    }
+
+    /// @notice Setting a max gas price for transactions. Can be set by the
+    ///         owner only.
+    /// @param _maxGasPrice Max gas price used to reimburse tx submitters.
+    function setMaxGasPrice(uint256 _maxGasPrice) external onlyOwner {
+        maxGasPrice = _maxGasPrice;
+
+        emit MaxGasPriceUpdated(_maxGasPrice);
+    }
+
+    /// @notice Withdraws all ETH from this pool which are sent to a given
+    ///         address. Can be set by the owner only.
+    /// @param receiver An address where ETH is sent.
+    function withdrawAll(address receiver) external onlyOwner {
+        withdraw(address(this).balance, receiver);
+    }
+
+    /// @notice Withdraws ETH amount from this pool which are sent to a given
+    ///         address. Can be set by the owner only.
+    /// @param amount Amount to withdraw from the pool.
+    /// @param receiver An address where ETH is sent.
+    function withdraw(uint256 amount, address receiver) public onlyOwner {
```

Comment:
I'm not sure if there's much value to it because:
- we have `requires` that will tell if the transaction fails, especially this one: `require(sent, "Failed to send Ether")`
- it will be executed by the gov which can see right away if the funds were received at the provided address
- it adds extra cost

Unless you have a specific case in mind when it's needed?

Reply:
It can be useful for maintainers/operators to know that some movement happened in the contract and there is a risk that transactions may not be reimbursable. 
`withdraw` is not a transaction that is expected to be executed often, so I don't think we should be worried about adding an event here.

Reply:
[b2781e1](https://github.com/keep-network/keep-core/pull/2826/commits/b2781e16230fd4935ec19337a6b5cd5792a60526)